start_admin_sshd.sh

```diff
@@ -47,6 +47,26 @@ get_user_data_keys() {
     ( IFS=$'\n'; echo "${valid_keys[*]}" )
 }
 
+# Export proxy environment variables for all users' login shells
+# Match the values of the proxy environment variables given to the admin container
+set_up_proxy_env_vars() {
+  PROXY_ENV_VAR_SETUP_SCRIPT="/etc/profile.d/br-proxy-env.sh"
+  local export_https_proxy
+  local export_no_proxy
+  if [[ ! -z ${HTTPS_PROXY:+x} ]]; then
+    export_https_proxy="export HTTPS_PROXY=${HTTPS_PROXY} https_proxy=${HTTPS_PROXY}"
+  fi
+  if [[ ! -z ${NO_PROXY:+x} ]]; then
+    export_no_proxy="export NO_PROXY=${NO_PROXY} no_proxy=${NO_PROXY}"
+  fi
```

Comment:
The if-present substitution seems redundant - directly checking the non-empty state is clearer here:

```suggestion
  if [[ ! -z "$HTTPS_PROXY" ]]; then
    export_https_proxy="export HTTPS_PROXY=${HTTPS_PROXY} https_proxy=${HTTPS_PROXY}"
  fi
  if [[ ! -z "$NO_PROXY" ]]; then
    export_no_proxy="export NO_PROXY=${NO_PROXY} no_proxy=${NO_PROXY}"
  fi
```

Reply:
`-n` is also recommended by [SC2236](https://github.com/koalaman/shellcheck/wiki/SC2236)

Reply:
I wanted to differentiate between the environment variable being unset, and being set but zero-length. I don't want to export any of the proxy variables if they were unset.

Reply:
See https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
> ${parameter:+word}
 >  If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.


`-n` can't check unset values, it'll always return true. Since it'll expand to just `[ -n ];`

The desired effect is for `test` to evaluate to true if the environment is _set_ (not necessary length > 0)

Reply:
This is no longer necessary if I use `declare -x` instead of exports as described in https://github.com/bottlerocket-os/bottlerocket-admin-container/pull/24#discussion_r605264971

Reply:
> -n can't check unset values, it'll always return true. Since it'll expand to just [ -n ];

This doesn't apply to `[[`. One of the reasons why I'm a proponent of `[[` 😄 .